### ISSUE TITLE
[FEATURE] Support drag-and-drop to load .nam models

### ISSUE THREAD

**Issue by jamesWalker55** [sdatkinson/NeuralAmpModelerPlugin#445]
**Is your feature request related to a problem? Please describe.**
In the plugin, the only way to load a model is to use the OS's built-in file selector. However, the location it initially shows seem to reset occasionally. I have to navigate all the way to the folder where I store my *.nam models again whenever this happens.

**Describe the solution you'd like**
Allow dragging and dropping *.nam models directly onto the plugin, e.g. from a file explorer or the DAW's sample browser.

Most DAWs like FL Studio or Reaper have a built in sample/file explorer and allow you to add 'favourite' folders. It's much easier to browse for .nam models there rather than the OS's built-in file selector, so I would like to suggest allowing dragging directly from an outside source to the plugin.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by DivannKokos** [comment]
at least in windows there are pinned folders in file explorer
and os's built-in file selector in windows is file explorer

**Comment by sdatkinson** [comment]
Can possibly implement using [`IControl.OnDrop()`](https://github.com/iPlug2/iPlug2/blob/678ae2d163c04fb1331cb39d1ca5f61b15b8af21/IGraphics/IControl.h#L144).

**Comment by synchu** [comment]
Right, I did a few quick changes here and there and it appears to work (at least on Windows, I have no means to test it with Mac).

@sdatkinson - I can see there's a Pull request that has been posted, so not sure how to proceed? Do I put a separate Pull request or try to get along with this one (not quite sure of my skills with Github :) )...

**Comment by sdatkinson** [comment]
@synchu you can start by pushing your changes to a branch on your fork and share about it here. 

I (or others) can have a look if I find time and handle the PR. But if you do that then anyone else can merge your changes into a branch of their own 👍

**Comment by synchu** [comment]
Thanks @sdatkinson. I applied the changes here:

https://github.com/synchu/NeuralAmpModelerPlugin

Now there's a required change into IControl.h, which I am not sure got committed, but it looks like these.
From:
` virtual void OnDrop(const char* str) {}`
to
` virtual void OnDrop(const char* str)
 {
   if (OnDropFunc)
   {
     OnDropFunc(str);
   }
 };

 std::function<void(const char* str)> OnDropFunc;`

**Comment by synchu** [comment]
Besides this suggested implementation, I wrote myself a "NAM Model Manager" that provides features for organizing, tagging, grouping by metadata and tags of NAM models. Besides that one can drag and drop an item from it to the NAM plugin. It does not store the .nam files, just path to the files and required metadata.
It is a bit bulky, being electron app, but does the job. I have the means to compile it for Windows only, but in theory it should work on Mac (and possibly Linux).

I added a standalone package .exe and a "setup" executable in the same repository with the my mod for NAM plugin that supports drag-and-drop.

https://github.com/synchu/NeuralAmpModelerPlugin/releases/tag/0.7.13-pre-draganddrop

If the drag-and-drop gets included in the NAM plugin or any other app by other vendors, supporting drag-and-drop, I will release it to the public.

All feedback is welcome.

**Comment by jamesWalker55** [comment]
Bumping since there seems to be more commit activity lately
